### PR TITLE
Let get_missed_upgrade_jobs.py support y and z version upgrade

### PR DESCRIPTION
**Pull request by JianLi-RH**
Test it on local machine:

z version:
```
[jianl@jianl-thinkpadt14gen4 ~]$ e2e_yaml=/home/jianl/1_code/release/ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml
[jianl@jianl-thinkpadt14gen4 ~]$ upgrade_yaml=/home/jianl/1_code/release/ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly-4.13-upgrade-from-stable-4.13.yaml
[jianl@jianl-thinkpadt14gen4 ~]$ 
[jianl@jianl-thinkpadt14gen4 ~]$ python /home/jianl/1_code/release/ci-operator/config/openshift/openshift-tests-private/tools/get_missed_upgrade_jobs.py $e2e_yaml $upgrade_yaml
- as: aws-c2s-ipi-disc-priv-fips-f28
- as: aws-ipi-disc-priv-mtu-localzone-fips-f28
- as: aws-ipi-disc-priv-sno-basecap-none-f28
- as: aws-ipi-disc-priv-sts-efs-f28
...
- as: vsphere-ipi-proxy-tp-f28
- ...
```

y version:
```
[jianl@jianl-thinkpadt14gen4 ~]$ e2e_yaml=/home/jianl/1_code/release/ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml
[jianl@jianl-thinkpadt14gen4 ~]$ upgrade_yaml=/home/jianl/1_code/release/ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly-4.13-upgrade-from-stable-4.12.yaml
[jianl@jianl-thinkpadt14gen4 ~]$ python /home/jianl/1_code/release/ci-operator/config/openshift/openshift-tests-private/tools/get_missed_upgrade_jobs.py $e2e_yaml $upgrade_yaml 'y'
- as: aws-ipi-disc-priv-sts-efs-f28
- as: aws-ipi-private-shared-vpc-phz-sts-f28
- as: aws-sc2s-ipi-disc-priv-fips-f28
- as: aws-usgov-ipi-disc-priv-f28
- as: azure-ipi-vmgenv1-f28
- as: azure-mag-ipi-fullyprivate-f28
- as: azure-stack-upi-f28
- as: baremetalds-ipi-ovn-dualstack-fips-f28
- as: baremetalds-ipi-ovn-ipv6-fips-f28
- as: gcp-ipi-basecap-none-additionalcaps-f28
- as: gcp-ipi-disk-encryption-f28
- as: gcp-ipi-mini-perm-custom-type-f28
- as: gcp-ipi-ovn-mtu-migrate-f28
- as: gcp-ipi-proxy-etcd-encryption-f28
- as: gcp-ipi-sdn-f28
- as: gcp-ipi-sdn-migration-ovn-f28
- as: gcp-ipi-xpn-techpreview-fips-f28
- as: gcp-upi-private-xpn-ingress-glb-f28
- as: gcp-upi-private-xpn-sdn-f28
- as: gcp-upi-xpn-f28
- as: vsphere-ipi-ovn-etcd-encryption-f28
- as: vsphere-ipi-ovn-local-to-shared-gateway-mode-migration-f28
- as: vsphere-ipi-ovn-shared-to-local-gateway-mode-migration-f28
- as: vsphere-ipi-sdn-migration-ovn-f28
[jianl@jianl-thinkpadt14gen4 ~]$ 
```

/cc @shellyyang1989 @liangxia @jianlinliu @evakhoni @dis016 @jiajliu

### CODE REVIEW

**Comment by JianLi-RH** [comment]
Unknown mode:
```
[jianl@jianl-thinkpadt14gen4 ~]$ python /home/jianl/1_code/release/ci-operator/config/openshift/openshift-tests-private/tools/get_missed_upgrade_jobs.py $e2e_yaml $upgrade_yaml 'a'
Unknown upgrade mode, available mode are 'y' and 'z'
[jianl@jianl-thinkpadt14gen4 ~]$
```

**Comment by JianLi-RH** [comment]
Default upgrade is 'y' can make sure we never add tech-preview jobs even it is indeed a z version upgrade.

**Comment by shellyyang1989** [comment]
/lgtm

cc @liangxia for review. Thank you!

File: ci-operator/config/openshift/openshift-tests-private/tools/get_missed_upgrade_jobs.py [hunk 1]
```diff
@@ -36,13 +37,14 @@ def get_jobs(yaml_file):
...
 
-def get_missed_profiles(e2e_jobs, upgrade_jobs):
+def get_missed_profiles(e2e_jobs, upgrade_jobs, mode='y'):
     """
     Get missed e2e jobs in upgrade jobs
     
     Args:
         e2e_jobs: e2e jobs
         upgrade_jobs: existing upgrade jobs
+        mode: upgrade mode, 'y' or 'z'
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/tools/get_missed_upgrade_jobs.py:47]: How about use `ystream`, `zstream` to make it more clear ?

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/tools/get_missed_upgrade_jobs.py:47]: Not a blocker.  Just let me know if you prefer the current names.

Comment (JianLi-RH) [reply ci-operator/config/openshift/openshift-tests-private/tools/get_missed_upgrade_jobs.py:47]: ystream, zstream will let us input more characters when run the script...I want it to be easier for users to run scripts.

**Pull request merged**